includes/class-wc-subscriptions-change-payment-gateway.php

```diff
@@ -434,6 +438,9 @@ public static function update_all_payment_methods_from_subscription( $subscripti
 				continue;
 			}
 
+			// Clear any stale _delayed_update_payment_method_all meta existing on the users other subscriptions if it exists.
+			$user_subscription->delete_meta_data( '_delayed_update_payment_method_all' );
```

Comment:
Subscription will be saved in the follow up `update_payment_method()` call or if not there, down further on line 447. No need to call save here.